File: api/public/models/project.py
from typing import List, Optional
from pydantic import BaseModel

class Link(BaseModel):
    name: str
    url: str

class Project(BaseModel):
    id: str
    email: str
    title: str
    description: str
    image: bytes
    tags: List[str]
    links: List[Link]
    index: int

class Projects(BaseModel): 
    projects: Optional[List[Project]]

class AddProjectInput(BaseModel):
    email: str
    title: str
    description: str
    image: bytes
    tags: List[str]
    links: List[Link]

class UpdateProjectInputPayload(BaseModel):
    title: Optional[str]
    description: Optional[str]
    image: Optional[bytes]
    tags: Optional[List[str]]
    links: Optional[List[Link]]
    index: Optional[int]

class DeleteProjectOutput(BaseModel):
    id: str

File: frontend/apps/salvatoreemilio.it/src/app/models/project.ts
export interface GetProjectsOutput {
  projects: Project[] | undefined;
}

export interface Project {
  id: string;
  email: string;
  title: string;
  description?: string;
  image?: string;
  tags?: string[];
  links?: Link[];
  index: number;
}

export interface Link {
  name: string;
  url: string;
}

export interface ProjectFeed {
  new_val?: Project;
  old_val?: Project;
}

export interface AddProjectInput {
  email: string;
  title: string;
  description?: string;
  image?: string;
  tags?: string[];
  links?: Link[];
}

export interface UpdateProjectInputPayload {
  title?: string;
  description?: string;
  image?: string;
  tags?: string[];
  links?: Link[];
}

export interface UpdateProjectInput {
  id: string;
  payload: UpdateProjectInputPayload;
}
